app/controllers/dogs_controller.rb

```diff
@@ -4,7 +4,13 @@ class DogsController < ApplicationController
   # GET /dogs
   # GET /dogs.json
   def index
-    @dogs = Dog.all
+    # set default page to 0 for root page
+    params["page"] == nil ? @page = 0 : @page = params["page"].to_i
```

Comment:
You can actually do `@page = params["page"].nil? ? 0 : params["page"].to_i`
A little nicer looking.